{{cookiecutter.repo_name}}/devtools/create_dev_env.py

```diff
@@ -0,0 +1,124 @@
+import os
+import argparse
+from pathlib import Path
+from conda.common.serialize import yaml_load
+from conda_env.pip_util import pip_args
+import subprocess as sp
+try:
+    from conda_build import api
+    from conda_build.environ import create_env
+    from conda_build.config import Config
+    from conda_build.conda_interface import url_path
+except ImportError as e:
+    e.msg += "\nNote:\n"
+    e.msg += "-------\n"
+    e.msg += "\nYou need to make sure that 'conda-build' is installed:"
+    e.msg += "\n    hint ---> `conda install conda-build`"
+    raise
+
+def ask_yn():
+    def question():
+        prompt = "Continue? [Y/n]"
+        try:
+            return raw_input(prompt).lower() or 'y'
+        except KeyboardInterrupt:
+            print('')
+            return 'n'
+        except NameError:
+            return input(prompt).lower() or 'y'
+
+    answer = question()
+    while not answer.startswith(('y', 'n')):
+        print("{} is not valid, please answer 'yes' or 'no'")
+        answer = question()
+    if answer.startswith('n'):
+        return False
+    else:
+        return True
+
+
+parser = argparse.ArgumentParser(
+    description=
+    "Creates a conda environment with the specified python version, and the run/test requirements of the provided recipe"
+)
+
+parser.add_argument("-n", "--name", type=str, help="The name of the environment, existing environments will be updated")
+parser.add_argument("-p", "--python", type=str, help="The version of python to be installed in environment")
+parser.add_argument("-c", "--channel", type=str, nargs="+", help="Additional channel(s) to search for packages (searched in order given, before defaults)")
+parser.add_argument("-y", "--yes", dest='ask', action='store_false')
+parser.add_argument("--file", type=str, help="An environment file (yaml) from which additional packages will be taken")
+parser.add_argument("recipe_path", type=str, help="The path to the recipe that will be used")
+
+parser.set_defaults(ask=True)
+
+
+def setup_dev_env(args):
+    # grab metadata (assume a simple setup with one build variant)
+
+    print(args.__dict__)
```

Comment:
Can we format this a bit rather than printing the raw dict or is this scratch?

Reply:
Sorry, scratch that did not get removed. I will make a cleanup pass over it